app/views/favorites/of_user.ctp

```diff
@@ -26,9 +26,21 @@
  */
 
 $username = Sanitize::paranoid($username, array("_"));
-$title = format(__("{user}'s favorite sentences", true), array('user' => $username));
+
+if ($userExists === true) {
+    $numberOfSentences = (int) $paginator->counter(
+        array(
+            "format" => "%count%"
+        )
+    );
+
+    $title = format(__("{user}'s favorite sentences", true), array('user' => $username));
+} else {
+    $title = format(__("There's no user called {username}", true), array('username' => $username));
+}
+
 $this->set('title_for_layout', $pages->formatTitle($title));
-$numberOfSentences = count($favorites);
+$numberOfSentences = $data;
```

Comment:
`$numberOfSentences` already contains the number of sentences, it is fetched from `$paginator` above (although not initialized to zero). You don’t need to use `$data` at all. You may remove it from the controller as well.